src/main/java/seedu/contax/logic/parser/BatchCommandParser.java

```diff
@@ -19,19 +21,32 @@
     public BatchCommand parse(String args) throws ParseException {
         requireNonNull(args);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_SEARCH_TYPE, PREFIX_EQUALS);
-
+                ArgumentTokenizer.tokenize(args,
+                        PREFIX_SEARCH_TYPE, PREFIX_EQUALS, PREFIX_START_WITH, PREFIX_END_WITH);
         String commandInput = argMultimap.getPreamble();
-        if (!argMultimap.arePrefixesPresent(PREFIX_SEARCH_TYPE, PREFIX_EQUALS)
-            || argMultimap.getPreamble().isEmpty()) {
+        String equalsText = "";
+        String startText = "";
+        String endText = "";
+        if (!argMultimap.arePrefixesPresent(PREFIX_SEARCH_TYPE) || argMultimap.getPreamble().isEmpty()
+            || (!argMultimap.arePrefixesPresent(PREFIX_EQUALS)
+                && !argMultimap.arePrefixesPresent(PREFIX_START_WITH)
+                && !argMultimap.arePrefixesPresent(PREFIX_END_WITH))) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
                     BatchCommand.MESSAGE_USAGE));
         }
-
+        if (argMultimap.arePrefixesPresent(PREFIX_EQUALS)) {
+            equalsText = argMultimap.getValue(PREFIX_EQUALS).get();
+        }
+        if (argMultimap.arePrefixesPresent(PREFIX_START_WITH)) {
+            startText = argMultimap.getValue(PREFIX_START_WITH).get();
+        }
+        if (argMultimap.arePrefixesPresent(PREFIX_END_WITH)) {
+            endText = argMultimap.getValue(PREFIX_END_WITH).get();
+        }
```

Comment:
You can check the optional from `getValue()` directly instead of using arePrefixesPresent.